bin/cdklocal

```diff
@@ -451,6 +451,13 @@ const patchPre_2_14 = () => {
   applyPatches(provider, CdkToolkit, SDK, ToolkitInfo);
 };
 
+const configureEnvironment = () => {
+  // This _must_ use localhost.localstack.cloud as we require valid subdomains of these paths to
+  // resolve. Unfortunately though `curl` seems to support subdomains of localhost, the CDK does not.
```

Comment:
So `s3.localhost` didn't work for you with cdklocal but `s3.localhost.localstack.cloud` did? :thinking:

Reply:
Yup unfortunately. I think it depends on the dns resolver and tool making the dns query, so sometimes it works and sometimes it doesn't depending on the tool.